### PR TITLE
Nelder mead V1.01

### PR DESCRIPTION
**Pull request by desmonty**
Here it is, better than before ! :man_dancing:
Again, if you see something odd, let me know.
I'm gonna finish the random_search (make unittest fast and convergence free).
After that we could continue freely and happily on anything.

### CODE REVIEW

File: source/Optimizers/nelder_mead.d [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+                 T expension_coef=2.0,
+                 T contraction_coef=0.5,
+                 T shrink_coef=0.5) {
+    return nelder_mead!T(_v, toDelegate(_obj),
+                         reflection_coef, expension_coef,
+                         contraction_coef, shrink_coef);
+}
+
+T nelder_mead(T)(ref Vector!T _v, T delegate(in Vector!T) _obj,
+                 in T _random_bound=5.0,
```
[the remark this answers is not in the document]
Comment (desmonty) [reply source/Optimizers/nelder_mead.d]: If you are talking about the coef, they came from the intial Paper and are the same in the sklearn implementation. However for the random bounds and maxIterations you're right, their implementation is interesting (for the maxter)
But I don't really like the implementation of the simplex. they seems to take a point, and create a simplex by modifying one axis per point on the simplex.

And they have no way to scale the simplex so _random_bound don't really have an equivalent parameter in their implementation. Also, in our implementation, we center the simplex around the origin. I made this choice because it seems a reasonable expectation in the special case of Neural Networks no ?

File: source/Optimizers/nelder_mead.d [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+    assert(_contraction_coef <= 0.5, "Contraction coef must be <= 0.5");
+    assert(_contraction_coef > 0, "Contraction coef must be > 0");
+
+    immutable size_t length = _v.length;
+    immutable size_t num_points = _v.length + 1;
+
+    /// Create Simplex
+
+    auto simplex = new Vector!T[num_points];
+    /+
```
[the remark this answers is not in the document]
Comment (desmonty) [reply source/Optimizers/nelder_mead.d]: mmh it is weird, it just think the all file changed. How can I fix it ?
